plugins/module_utils/ec2.py

```diff
@@ -805,3 +805,86 @@ def compare_aws_tags(current_tags_dict, new_tags_dict, purge_tags=True):
             tag_key_value_pairs_to_set[key] = new_tags_dict[key]
 
     return tag_key_value_pairs_to_set, tag_keys_to_unset
+
+
+def _describe_ec2_tags(client, **params):
+    paginator = self.client.get_paginator('describe_tags')
+    return paginator.paginate(Filters=filters).build_full_result()
+
+
+def describe_ec2_tags(client, module, resource_id, resource_type=None, retry_codes=[]):
+    """
+    Performs a paginated search of EC2 resource tags.
+    """
+    filters = {'resource-id': resource_id}
+    if resource_type:
+        filters['resource-type'] = resource_type
+    filters = ansible_dict_to_boto3_filter_list(filters)
+
+    try:
+        results = AWSRetry.jittered_backoff(
+            retries=10,
+            catch_extra_error_codes=retry_codes
+        )(
+            _describe_ec2_tags
+        )(
+            client, Filters=filters
+        )
```

Comment:
Nit but I really don't like the way this is (auto?) formatted.

Reply:
Manually, trying to avoid really long lines